californium-core/src/main/java/org/eclipse/californium/core/network/BaseMatcher.java

```diff
@@ -184,7 +187,6 @@ protected final Exchange matchNotifyResponse(final ExchangeObserver exchangeObse
 			request.setDestinationPort(response.getSourcePort());
 			exchange = new Exchange(request, Origin.LOCAL, obs.getContext());
 			exchange.setRequest(request);
-			exchange.setObserver(exchangeObserver);
```

Comment:
I'm not sure this is not usefull. Don't we need `exchangeObserver` for blockwise ?